syntax="proto3";

package helloworld;

message GetDeptUserRequest{
  uint32 dept_id = 1;
  string dept_name=2;
  repeated uint32  uid_list = 3;
  map<string,string> filter = 4;
}
message BasicUser{
  uint32 id=1;
  string name=2;
}
message GetDeptUserResponse{
  repeated BasicUser user_list=1;
  map<uint32,BasicUser> user_map=2;
}



service Greeter{
  rpc SayHello (HelloRequest) returns (HelloResponse) {}
  rpc GetDeptUser (GetDeptUserRequest) returns (GetDeptUserResponse) {}
}

message HelloRequest{
  string name=1;
  int32 age=2;
}
message HelloResponse{
  string message=1;
}